src/teleport.cpp

```diff
@@ -70,13 +70,36 @@ void Teleport::addThing(Thing* thing)
 	return addThing(0, thing);
 }
 
+bool Teleport::checkInfinityLoop(Tile* destTile)
+{
+	if (!destTile) {
+		return false;
+	}
+
+	if (Teleport* teleport = destTile->getTeleportItem()) {
+		const Position& nextDestPos = teleport->getDestPos();
+		if (getPosition() == nextDestPos) {
+			return true;
+		}
+		return checkInfinityLoop(g_game.map.getTile(nextDestPos));
+	}
+	return false;
+}
+
 void Teleport::addThing(int32_t, Thing* thing)
 {
 	Tile* destTile = g_game.map.getTile(destPos);
 	if (!destTile) {
 		return;
 	}
 
+	// Prevent infinity loop (anti fools)
```

Comment:
I think we could get rid of that (anti fools) just leave it as prevent infinity loop